base/meta.jl

```diff
@@ -223,6 +223,55 @@ function parse(str::AbstractString; raise::Bool=true, depwarn::Bool=true)
     return ex
 end
 
+"""
+    parsefile(filename; raise=true, depwarn=true)
+
+Parse the Julia file located at `filename`.
+Returns a vector that contains all expressions in the file.
+Throws an `IOError` if the file `filename` does not exist.
+If `raise` is `true` (default),
+syntax errors will raise an error; otherwise, `parse` will return an expression that will
+raise an error upon evaluation.  If `depwarn` is `false`, deprecation warnings will be
+suppressed.
+
+```julia
+julia> Meta.parsefile("foo.jl")
+```
+"""
+function parsefile(filename::AbstractString;
+                   raise::Bool=true,
+                   depwarn::Bool=true)::Vector{Union{Expr, LineNumberNode}}
+    if !ispath(filename)
+        throw(IOError("The file \"$(filename)\" does not exist"))
+    end
+    full_file_path = realpath(filename)::String
+    file_contents = read(full_file_path, String)::String
+    block = Meta.parse("""
+                       begin
+                           $(file_contents)
+                       end
+                       """;
+                       raise = raise,
+                       depwarn = depwarn)::Expr
+    block_fixed = _parsefile_fix_filename(block,
+                                          full_file_path)::Expr
+    lines = convert(Vector{Union{Expr, LineNumberNode}}, block_fixed.args)::Vector{Union{Expr, LineNumberNode}}
+    return lines
+end
+
+@inline function _parsefile_fix_filename(expr::Expr,
+                                         correct_filename::String)::Expr
+    return Expr(expr.head,
+                _parsefile_fix_filename.(expr.args, correct_filename)...)
+end
+
+@inline function _parsefile_fix_filename(lnn::LineNumberNode,
+                                         correct_filename::String)::LineNumberNode
+    return LineNumberNode(lnn.line, correct_filename)
```

Comment:
This should probably be `lnn.line-1`, to adjust for the `begin` keyword that surrounds things.

Reply:
Done!

Reply:
@davidanthoff I removed the newline after `begin`. I think that means we don't need to subtract `1` from the line number. Am I correct?

Reply:
I would assume so!

Reply:
If you get a chance, test it out.

Reply:
I'm not setup to compile Julia itself...